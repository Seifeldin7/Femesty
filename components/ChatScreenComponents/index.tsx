export * from './ChatInputBox'
export * from './ChatMessage'