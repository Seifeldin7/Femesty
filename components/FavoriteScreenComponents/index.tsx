export * from "./FavoriteHzScroller";
export * from "./FavoriteLook";
