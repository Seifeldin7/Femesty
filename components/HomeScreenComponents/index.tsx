export * from "./FilterButton"
export * from "./HeaderTab"
export * from "./HorizontalScroller"
export * from "./LookMedia"
export * from "./SideBar"