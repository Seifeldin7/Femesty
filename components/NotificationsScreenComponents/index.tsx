export * from "./HiddenItem";
export * from "./VisibleItem";
