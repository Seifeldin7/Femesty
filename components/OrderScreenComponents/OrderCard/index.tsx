import React from "react";
import { View, Text, Image } from "react-native";
import { FontAwesome } from "@expo/vector-icons";
import * as Animatable from "react-native-animatable";

import { styles, mystateStyle } from "./styles";
import { componentProps } from "./types";
import { MainColors } from "../../../styles/Colors";
import { Avatar } from "../../ThemeComponents/Avatar";

export const OrderCard = ({ order, TestId }: componentProps) => {
  return (
    <Animatable.View
      animation="fadeInRightBig"
      duration={1000}
      delay={200}
      style={{ backgroundColor: MainColors.Natural }}
    >
      <View style={styles.container} testID={TestId}>
        <View style={styles.image}>
          {order.product.image1 !== "" ? (
            <Avatar style={styles.logo} imgPath={{ uri: order.product.image1 }} size="medium" />
          ) : (
            <FontAwesome name="user-circle-o" size={70} color="gray" />
          )}
        </View>
        <View style={styles.cardContent}>
          <View style={styles.cardData}>
            <View style={styles.productDetails}>
              <Text numberOfLines={2} style={styles.title}>
                {order.product.name}
              </Text>
              <Text numberOfLines={1} style={styles.brand}>
                {order.product.seller.name}
              </Text>
              <View style={styles.sizeColor}>
                <Text numberOfLines={1} style={styles.size}>
                  size: <Text style={styles.actualSize}>{order.size}</Text>
                </Text>
                <View
                  style={[styles.color, { backgroundColor: order.color }]}
                ></View>
              </View>
            </View>
            <View style={styles.divider} />
            <View style={styles.statePrice}>
              <Text
                style={[styles.state, { color: mystateStyle[order.status] }]}
              >
                {order.status}
              </Text>
              <Text numberOfLines={1} style={styles.price}>
                {order.price.toFixed(2)} <Text style={styles.egp}>EGP</Text> 
              </Text>
            </View>
          </View>
          <View style={styles.date}>
            <Text numberOfLines={1} style={styles.TextDate}>
              requested on {order.created_at.toString()}
            </Text>
          </View>
        </View>
      </View>
    </Animatable.View>
  );
};
