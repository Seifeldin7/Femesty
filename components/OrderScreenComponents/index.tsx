export * from './OrderItem'
export * from './OrderCard'
export * from './OrderList'