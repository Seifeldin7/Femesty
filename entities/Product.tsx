export default class Product {
    public id!: number;
    public name: string;
    public description: string;
    public price: number;
    public sizes: string[];
    public colors: string[];
    public image1: any;
    public image2: any;
    public image3: any;
    public category: any;
    public seller: any;
    public created_at: Date;
  
    constructor(
      id: number,
      name: string,
      description: string,
      price: number,
      sizes: string[],
      colors: string[],
      image1: any,
      image2: any,
      image3: any,
      category: any,
      seller: any,
      created_at: Date
    ) {
      this.id = id;
      this.name = name;
      this.description = description;
      this.price = price;
      this.image1 = image1;
      this.image2 = image2;
      this.image3 = image3;
      this.sizes = sizes;
      this.category = category;
      this.created_at = created_at;
      this.seller = seller;
      this.colors = colors;
    }
  }
  