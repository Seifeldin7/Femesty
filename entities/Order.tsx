import Product from "./Product";

export default class Order {
    public  id!: number;
    public  status: string;
    public  price: number;
    public  shipping_price: number;
    public  created_at: Date;
    public  color: string;
    public  size: string;
    public  order_number: number;
    public  product: Product
    constructor(
        id: number,
        price: number,
        shipping_price: number,
        created_at: Date,
        color: string,
        size: string,
        order_number: number,
        status: string,
        product: Product
        ) {

        this.id = id,
        this.shipping_price = shipping_price;
        this.price = price;
        this.created_at = created_at;
        this.color = color;
        this.size = size;
        this.order_number = order_number;
        this.status = status;
        this.product = product;
    }
}