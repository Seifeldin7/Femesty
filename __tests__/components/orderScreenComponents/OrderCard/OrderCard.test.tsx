import React from "react";
import { Text } from "react-native";
import { render, fireEvent } from "react-native-testing-library";
import { ReactTestInstance } from "react-test-renderer";
import { Alert } from "react-native";

import { OrderCard } from "../../../../components/OrderScreenComponents";
import Order from "../../../../entities/Order";
import Product from "../../../../entities/Product";

test("should render Selected Order List in all screens", () => {
  const tagName = "Hello";
  let deleteOrderHandler = jest.fn();
  const order = new Order(
    1,
    30,
    299,
    new Date(),
    "red",
    "XL",
    1,
    "Accepted",
    new Product(
      1,
      "T-Shirt",
      "verey elegent",
      299,
      ["x", "xl", "m"],
      ["red", "white", "blue"],
      "../../../../assets/images/02.jpg",
      "../../../../assets/images/02.jpg",
      "../../../../assets/images/02.jpg",
      "1",
      "Zara",
      new Date()
    )
  );
  const tree = render(<OrderCard order={order} TestId="OrderList" />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("<OrderListItems>", () => {
  let deleteOrderHandler: (value?: string | undefined) => void,
    getByTestId: (text: string) => ReactTestInstance;

  beforeEach(() => {
    deleteOrderHandler = jest.fn();
    const order = new Order(
        1,
        30,
        299,
        new Date(),
        "red",
        "XL",
        1,
        "Accepted",
        new Product(
          1,
          "T-Shirt",
          "verey elegent",
          299,
          ["x", "xl", "m"],
          ["red", "white", "blue"],
          "../../../../assets/images/02.jpg",
          "../../../../assets/images/02.jpg",
          "../../../../assets/images/02.jpg",
          "1",
          "Zara",
          new Date()
        )
      );
    ({ getByTestId } = render(
        <OrderCard order={order} TestId="OrderList" />
    ));
  });

  describe("Interactive test", () => {
    it("should call delete order on press", () => {
      expect(deleteOrderHandler).toBeDefined();
    });
    it("should render with right swipeout", async () => {
      expect(getByTestId("OrderList").props.children).toBeInstanceOf(Object);
    });

    it("should call render amd all props defined", async () => {
      expect(getByTestId("OrderList").props).toBeDefined();
      expect(getByTestId("OrderList").children).toBeDefined();
    });
  });
});