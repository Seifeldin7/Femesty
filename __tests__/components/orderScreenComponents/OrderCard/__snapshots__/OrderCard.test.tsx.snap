// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Selected Order List in all screens 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#dbddde",
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {},
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      forwardedRef={[Function]}
      style={
        Object {
          "backgroundColor": "#F6F6F6",
          "opacity": 0,
          "transform": Array [
            Object {
              "translateX": 500,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 15,
            "flex": 1,
            "flexDirection": "row",
            "height": 266.8,
            "justifyContent": "space-between",
            "paddingHorizontal": 5,
            "width": 710,
          }
        }
        testID="OrderList"
      >
        <View
          style={
            Object {
              "flex": 1.1,
              "margin": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "../../../../assets/images/02.jpg",
              }
            }
            style={
              Object {
                "borderColor": "grey",
                "borderRadius": 35,
                "borderWidth": 2,
                "height": 70,
                "marginRight": 5,
                "marginTop": 5,
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "flex": 4,
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 6,
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 2.3,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginTop": 15,
                  }
                }
              >
                T-Shirt
              </Text>
              <Text
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flex": 1,
                    "fontSize": 20,
                    "fontWeight": "bold",
                  }
                }
              >
                Zara
              </Text>
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "flex": 1,
                      "fontSize": 20,
                    }
                  }
                >
                  size:
                  XL
                </Text>
                <View
                  style={
                    Array [
                      Object {
                        "borderRadius": 12.5,
                        "height": 25,
                        "width": 25,
                      },
                      Object {
                        "backgroundColor": "red",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1.7,
                  "justifyContent": "center",
                  "paddingVertical": 20,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    },
                    Object {
                      "color": "mediumaquamarine",
                    },
                  ]
                }
              >
                Accepted
              </Text>
              <Text
                style={
                  Object {
                    "alignSelf": "center",
                    "flex": 1,
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                30
                .00 EGP
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
              }
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 12,
                }
              }
            >
              requested on 
              Fri, 11 Dec 2020 02:40:07 GMT
               
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View />
  <View />
</View>
`;
